lib/logstash/plugin_mixins/normalize_config_support.rb

```diff
@@ -73,31 +79,151 @@ def with_deprecated_alias(deprecated_alias)
           with_deprecated_mapping(deprecated_alias) { |v| v }
         end
 
+        ##
+        # Map one or more configs as required. When the canonical config is set, all required aliases
+        # must have a value or default assigned.
+        #
+        # @raise [ArgumentError] if the required mappings is already set
+        # @raise [ArgumentError] if the required aliases argument is not provided
+        # @raise [ArgumentError] if any required alias does not exists
+        # @param required_aliases [String...]: the required param names
+        #
+        # @return [void]
+        #
+        def with_required_aliases(*required_aliases)
+          require_value!('required_aliases', required_aliases)
+          fail(ArgumentError, 'Canonical config cannot be mapped as a required alias') if required_aliases.include?(@canonical_config)
+          fail(ArgumentError, 'Required aliases mappings already configured for this config normalizer') unless @required_mapping.nil?
+          ensure_configs_exists!(required_aliases)
+
+          @required_mapping = required_aliases.map(&:to_s)
+        end
```

Comment:
In the spirit of the other suggested names, this could be `requires_valid`, so that we could have `@username = normalize_config(:username) {|c| c.requires_valid(:password) }`.